spec/requests/custom_actions_spec.rb

```diff
@@ -220,4 +220,471 @@ def expect_result_to_have_custom_actions_hash
       expect(response.parsed_body).to include(expected)
     end
   end
+
+  def define_custom_button1(resource)
+    dialog1          = FactoryGirl.create(:dialog, :label => "dialog1")
+    resource_action1 = FactoryGirl.create(:resource_action, :dialog_id => dialog1.id)
+    FactoryGirl.create(:custom_button, :name => "button1", :applies_to => resource, :userid => @user.userid, :resource_action => resource_action1)
+  end
+
+  describe "Availability Zones" do
+    before(:each) do
+      @resource = FactoryGirl.create(:availability_zone)
+      define_custom_button1(@resource)
+    end
+
+    it "queries return custom actions defined" do
+      api_basic_authorize(action_identifier(:availability_zones, :read, :resource_actions, :get))
+
+      get api_availability_zone_url(nil, @resource)
+
+      expect(response.parsed_body).to include(
+        "id"      => @resource.id.to_s,
+        "href"    => api_availability_zone_url(nil, @resource),
+        "actions" => a_collection_including(a_hash_including("name" => "button1"))
+      )
+    end
+
+    it "accept custom actions" do
+      api_basic_authorize
+
+      post api_availability_zone_url(nil, @resource), :params => gen_request(:button1, "key1" => "value1")
+
+      expect_single_action_result(:success => true, :message => /.*/, :href => api_availability_zone_url(nil, @resource))
+    end
+  end
+
+  describe "CloudSubnet" do
+    before(:each) do
+      @resource = FactoryGirl.create(:cloud_subnet)
+      define_custom_button1(@resource)
+    end
+
+    it "queries return custom actions defined" do
+      api_basic_authorize(action_identifier(:cloud_subnets, :read, :resource_actions, :get))
+
+      get api_cloud_subnet_url(nil, @resource)
+
+      expect(response.parsed_body).to include(
+        "id"      => @resource.id.to_s,
+        "href"    => api_cloud_subnet_url(nil, @resource),
+        "actions" => a_collection_including(a_hash_including("name" => "button1"))
+      )
+    end
+
+    it "accept custom actions" do
```

Comment:
NIT, but I think all of these should say "accepts". I'll have to update my PRs as well.